### PR TITLE
Downgrade expected stall detector warning to info

### PR DESCRIPTION
**Pull request by travisdowns**
On most distributions and containers the perf-based stall detector will fail because /proc/sys/kernel/perf_event_paranoid is set to 2 or higher, which prevents non-root processes from profiling the kernel.

Instead we log a different message, telling the user what they can do if they want kernel backtraces, at INFO and only on shard 0 (so we don't need to see this advice repeated 128 times on a big box).

### CODE REVIEW

File: src/core/reactor.cc [hunk 1]
```diff
@@ -1341,10 +1341,28 @@ cpu_stall_detector_linux_perf_event::try_make(cpu_stall_detector_config cfg) {
...
-        return cpu_stall_detector_linux_perf_event::try_make(cfg);
+        try {
+            return cpu_stall_detector_linux_perf_event::try_make(cfg);
+        } catch (std::system_error& e) {
+            // This failure occurs when /proc/sys/kernel/perf_event_paranoid is set
+            // to 2 or higher, and is expected since most distributions set it to that
+            // way as of 2023. In this case we log a different message and only at INFO
+            // level on shard 0.
+            was_eaccess_failure = e.code() == std::error_code(EACCES, std::system_category());
+            throw e;
```
Comment (avikivity) [inline src/core/reactor.cc:1354]: `throw;` is more accurate as it avoids slicing.

File: src/core/reactor.cc [hunk 2]
```diff
@@ -1341,10 +1341,28 @@ cpu_stall_detector_linux_perf_event::try_make(cpu_stall_detector_config cfg) {
...
+        }
     } catch (...) {
-        seastar_logger.warn("Creation of perf_event based stall detector failed, falling back to posix timer: {}", std::current_exception());
+        if (was_eaccess_failure) {
```
[the remark this answers is not in the document]
Comment (avikivity) [reply src/core/reactor.cc:1357]: I've never seen anything better.

File: src/core/reactor.cc [hunk 3]
```diff
@@ -1341,10 +1341,28 @@ cpu_stall_detector_linux_perf_event::try_make(cpu_stall_detector_config cfg) {
...
+            if (this_shard_id() == 0) {
+                seastar_logger.info("Perf-based stall detector creation failed (EACCESS), "
```
Comment (avikivity) [inline src/core/reactor.cc:1359]: There is `logger::info0()` just for this.

File: src/core/reactor.cc (hunk 1 above)
Comment (travisdowns) [reply src/core/reactor.cc]: How embarrassing. Fixed.

File: src/core/reactor.cc (hunk 3 above)
Comment (travisdowns) [reply src/core/reactor.cc]: Nice TIL. Fixed.

**Pull request merged**